sql/catalyst/src/main/scala/org/apache/spark/sql/catalog/v2/expressions/expressions.scala

```diff
@@ -111,32 +122,104 @@ private[sql] final case class ApplyTransform(
   override def toString: String = describe
 }
 
+/**
+ * Convenience extractor for any Literal.
+ */
+private object Lit {
+  def unapply[T](literal: Literal[T]): Some[(T, DataType)] = {
```

Comment:
`Some` -> `Option`?

Reply:
This always returns `Some` so I thought it was correct to use here. This practice was pointed out by @HyukjinKwon here: https://github.com/apache/spark/pull/24689#discussion_r287632404